client/src/pages/AddChallenge.tsx

```diff
@@ -111,26 +112,9 @@ function AddChallenge() {
         });
     }
 
-    const [userRole, setUserRole] = useState<string | null>(null); 
+    const {isAdmin } = useCheckRole();
```

Comment:
Looks good. so we are using the checkrole from hooks instead of redoing it in other pages